janusgraph-core/src/main/java/org/janusgraph/graphdb/database/management/ManagementSystem.java

```diff
@@ -175,7 +175,8 @@ public ManagementSystem(StandardJanusGraph graph, KCVSConfiguration config, Log
     private final UserModifiableConfiguration.ConfigVerifier configVerifier = new UserModifiableConfiguration.ConfigVerifier() {
         @Override
         public void verifyModification(ConfigOption option) {
-            Preconditions.checkArgument(option.getType() != ConfigOption.Type.FIXED, "Cannot change the fixed configuration option: %s", option);
+            Preconditions.checkArgument (graph.getConfiguration().getAllowUpgrade() && option.getName() == "storage-version" || 
```

Comment:
I was debating creating a separate function to check the option name in case we wanted to add other options, like `janusgraph-version`. I think it might be good to update `janusgraph-version` as well as the storage-version so that old clients that still have the compatible versions check will realize that the graph it's connecting to is not entirely compatible. 


The check for the option name would instead look like
`upgradableElement(option.getName())`
function would probably look like:
```
private boolean upgrableableElement(name) {
    private String[] upgradable = new String[] {"storage-version", "janusgraph-version"};
    return Arrays.asList(upgradeable).contains(name);
}
```

A similar function would have to be added to ModifiableConfiguration.java.